cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -53,29 +59,110 @@
 import org.junit.Before;
 import org.junit.ClassRule;
 import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.File;
 import java.io.IOException;
 import java.net.URI;
+import java.net.URL;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import javax.annotation.Nullable;
+import javax.ws.rs.ServiceUnavailableException;
 
 /**
  *
  */
 public class IntegrationTestBase {
+  private static final Logger LOG = LoggerFactory.getLogger(IntegrationTestBase.class);
 
   @ClassRule
   public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
 
   @Before
   public void setUp() throws Exception {
+    checkSystemServices();
+    assertUnrecoverableResetEnabled();
     assertIsClear();
   }
 
+  private void checkSystemServices() throws ServiceUnavailableException {
+    MonitorClient monitorClient = getMonitorClient();
+
+    try {
+      checkSystemServicesOk(monitorClient);
+    } catch (UnauthorizedException e) {
+      LOG.info("Caught UnAuthorizedException, Getting Credentials");
+      // security is enabled, we need to get access token before checking system services
+      URL url = null;
+      try {
+        Properties properties = new Properties();
+        properties.setProperty("security.auth.client.username", System.getProperty("userName"));
+        properties.setProperty("security.auth.client.password", System.getProperty("password"));
+        AuthenticationClient authClient = new BasicAuthenticationClient();
+        authClient.configure(properties);
+        ConnectionConfig connectionConfig = getClientConfig().getConnectionConfig();
+        authClient.setConnectionInfo(connectionConfig.getHostname(), connectionConfig.getPort(), false);
+        AccessToken accessToken = getAccessToken(authClient);
+        System.setProperty("accessToken", accessToken.getValue());
```

Comment:
It would be better to save the access token and return it in a method called `getAcccessToken()`. Also have a method called `protected AccessToken fetchAccessToken()`, that user can over-ride if required. We can have an implementation that takes in username and password and returns the access token.